src/deserialize/parse.rs

```diff
@@ -197,34 +197,79 @@ fn read_discard(chars: &mut std::iter::Enumerate<std::str::Chars>) -> Result<Opt
 }
 
 fn read_number(n: char, chars: &mut std::iter::Enumerate<std::str::Chars>) -> Result<Edn, Error> {
+    let delimiters = [',', ']', '}', ')', ';'];
     let i = chars
         .clone()
         .next()
         .ok_or_else(|| Error::ParseEdn("Could not identify symbol index".to_string()))?
         .0;
     let c_len = chars
         .clone()
-        .take_while(|c| {
-            c.1.is_numeric()
-                || c.1 == '.'
-                || c.1 == '/'
-                || c.1 == 'e'
-                || c.1 == 'E'
-                || c.1 == '+'
-                || c.1 == '-'
-        })
+        .take_while(|(_, c)| !c.is_whitespace() && !delimiters.contains(c))
         .count();
-    let mut number = String::new();
-    let string = chars.take(c_len).map(|c| c.1).collect::<String>();
-    number.push(n);
-    number.push_str(&string);
+    let (number, radix) = {
+        let mut number = String::new();
+        number.push(n);
+        for (_, c) in chars.take(c_len) {
+            number.push(c)
+        }
+        if number.to_lowercase().starts_with("0x") {
+            number.remove(0);
+            number.remove(0);
+            (number, 16)
+        } else if number.to_lowercase().starts_with("-0x") {
+            number.remove(1);
+            number.remove(1);
+            (number, 16)
+        } else if let Some(index) = number.to_lowercase().find('r') {
```

Comment:
like I sair previously, I'm keen on having the radix at the very end, as `number_r8`, easier to parse as well. As this is something completely new in edn, I would go with this